Source/Private/NewAzureServiceConnection.ps1

```diff
@@ -0,0 +1,129 @@
+function NewAzureServiceConnection {
+    param(
+        [string]$Organization,
```

Comment:
You are using parameterSets but haven't defined them in parameters or CmdLetBinding()
I am curious if this works at all, but even if it works without, please include both defaultParameterSet and Parameter bindings for clarity.

Reply:
If we add **cmdletbinding** to this then it will have its own scope. AFAIK if you invoke a regular function, it will inherit the cmdlet scope. In the example below you could remove `@PSBoundParameters` from `Test-NestedFunction` and notice that `$PSCmdlet` is still `Test-CmdletBinding`

```ps1
$VerbosePreference = 'Continue'

function Test-CmdletBinding {
    [CmdletBinding(DefaultParameterSetName = "ParamSet1")]
    param (
        [Parameter(Mandatory, ParameterSetName = "ParamSet1")]
        [string]
        $InputString,
        [Parameter(Mandatory, ParameterSetName = "ParamSet2")]
        [int]
        $InputInt
    )

    Write-Host "Test-CmdletBinding:  $($PSCmdlet.ParameterSetName)"
    Write-Host "--"
    Test-NestedCmdletBinding @PSBoundParameters
    Write-Host "--"
    Test-NestedFunction @PSBoundParameters
    Write-Host "--"
    Test-NestedFunction @PSBoundParameters
    Write-Host "--"
}

function Test-NestedCmdletBinding {
    [CmdletBinding(DefaultParameterSetName = "ParamSet3")]
    param (
        [Parameter(Mandatory, ParameterSetName = "ParamSet3")]
        [string]
        $InputString,
        [Parameter(Mandatory, ParameterSetName = "ParamSet4")]
        [int]
        $InputInt
    )

    Write-Host "Test-NestedCmdletBinding: $($PSCmdlet.ParameterSetName)"
}


function Test-NestedFunction {
    param (
        [string] $InputString,
        [int] $InputInt
    )

    Write-Host "Test-NestedFunction: $($PSCmdlet.ParameterSetName)"
}

Write-Host "========================================"
Test-CmdletBinding -InputString "Hello World"

Write-Host "========================================"
Test-CmdletBinding -InputInt 5
```

I'm open to changing it but it seems like it will be error prone no matter which direction we go. I'm going to try to learn more about this area of powershell this weekend.